src/plugins/eval.lua

```diff
@@ -0,0 +1,75 @@
+local eval = {}
+
+eval.help = 'eval <lua-expr>'
+
+eval.deepcopy = function (orig)
+    local orig_type = type(orig)
+    local copy
+    if orig_type == 'table' then
+        copy = {}
+        for orig_key, orig_value in next, orig, nil do
+            copy[eval.deepcopy(orig_key)] = eval.deepcopy(orig_value)
+        end
+        setmetatable(copy, eval.deepcopy(getmetatable(orig)))
+    else -- number, string, boolean, etc
+        copy = orig
+    end
+    return copy
+end
+
+eval.inspect = function (thing)
+    if type(thing) == 'table' then
+        local result = ''
+        for k, v in pairs(thing) do
+            if type(v) == 'table' then
```

Comment:
I believe this could be simplified quite a bit:

```lua
local formats = {
    table = '{…}',
    function = '<function>',
    string = '"%s"'
}

local value_format = formats[type(v)] or '%s'
result = ('%s, %s = ' .. value_format):format(result, k, v)
```

This manages to remove the giant if/else block in favor of a clearer explanation of what's happening (special-case printing).